### PR TITLE
Fix #76 fastcdc chunk boundary off-by-one.

### LINKED ISSUE #76: fastcdc implementation sets chunk boundaries before last gear-hashed byte.
The fastcdc implementation adds `data[i]` in the gear rollsum, checks if the hash meets the chunk boundary criteria, and then returns `i` as the chunk length. Since `data` is zero indexed, this means the last byte included in the rollsum that met the boundary criteria is not included on the end the chunk, but ends up at the start of the next chunk.

This has an interesting degenerative case; if you get all the identified chunks for a file and re-arrange them into a different order, then cdc-file-transfer will find nearly zero duplicate chunks, even though every single chunk is duplicated. The only duplicates it will find are the ones where the byte after the chunk happens to be the same in both files.

In practice this probably rarely happens, as changes would rarely align exactly on the edge of chunk boundaries, but it would be a trivial fix to avoid this degenerative case.

I don't know how important preserving the chunking behaviour is for backwards compatibility... are "signatures" of the chunk lengths/hashes stored and used later, possibly by different versions?

I can easily put together a pull request to fix this if backwards-compatibility is not a problem.

### PR DESCRIPTION
**Pull request by dbaarda**
This ensures that the last byte included in the gear-hash that identified the  chunk boundary is included in the chunk. This ensures chunks are still matched  when the byte immediately after them is changed.

It also initializes the gear hash to all 1's to avoid zero-length chunks when `min_size=0`. It changes the fastcdc `MaxChunkSize` test to use `min_size=0` to test this corner-case.

### CODE REVIEW

**Comment by dbaarda** [comment]
Note that the case of `min_size=0` is a special corner case because we never actually emit zero-length chunks. There were several different ways this could have been handled;

1. Add an `assert()` that checks `min_size >= 1` in the initializer. This would not catch the problem in optimized builds, and I don't know if there are already cases using `min_size=0`.
2. Change the "contract" to `min_size < chunk_size <= max_size` instead of  `min_size <= chunk_size <= max_size`, so we always emit chunks larger than `min_size`. This feels a little confusing/ugly somehow.
3. Just use `min_size=1` internally whenever `min_size=0` is specified in the config. This was a slightly bigger change than what I chose.
4. Initialize the gear hash to all 1's so that it's never a match before any bytes have been rolled in. The old init value of all 0's would always match, meaning we'd get stuck emitting zero-length chunks and never progress when `min_size=0`.

I went with option 4 because it was the smallest change that doesn't change the existing "contract" for how `min_size` is interpreted.

File: fastcdc/fastcdc.h [hunk 1]
```diff
@@ -239,7 +239,8 @@ class ChunkerTmpl {
       len = cfg_.max_size;
     }
 
-    uint64_t hash = 0;
+    // Init hash to all 1's to avoid zero-length chunks with min_size=0.
+    uint64_t hash = (uint64_t)-1;
```
Comment (ljusten) [inline fastcdc/fastcdc.h:243]: This option SGTM. In practice, we always choose min_size > 0, anyway, so it's a rather theoretical consideration where we should just make sure that things don't explode.

**Review by ljusten — APPROVED** [review]
Thanks!

File: fastcdc/fastcdc.h (hunk 1 above)
Comment (dbaarda) [reply fastcdc/fastcdc.h:243]: Agreed.
